### PR TITLE
Validate that reviewers are provided when review workflow enabled

### LINKED ISSUE #402: Collection review config: validate at least one reviewer is listed if Review Workflow is enabled
follow on from https://github.com/sul-dlss/happy-heron/issues/348 / https://github.com/sul-dlss/happy-heron/pull/384

validate that reviewers are listed if the reviewer workflow is enabled (@mjgiarlo suggests: "there are two strategies here: pure JavaScript validation or use remote form validation (how we're dealing with keywords)" -- see https://github.com/sul-dlss/happy-heron/pull/384#discussion_r520095850)

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

fixes #402

### CODE REVIEW

File: app/javascript/controllers/reviewer_sunets_controller.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller  {
+  static targets = ["container", "error"];
```
Comment (peetucket) [inline app/javascript/controllers/reviewer_sunets_controller.js:4]: i'm not seeing the container target defined in the html here...is it part of an already existing component?

**Review by peetucket — APPROVED** [review]
One question...

File: app/components/collections/form_component.html.erb [hunk 2]
```diff
@@ -94,11 +98,11 @@
 
     <!-- TODO: only display sunet entry field if reviewer workflow is enabled -->
     <p>Enter SUNet IDs of participants. Separate each one with a comma and a space. (e.g. janelath, lelandst)</p>
-    <div class="mb-3 row">
+    <div class="mb-3 row" data-controller="reviewer-sunets" data-action="error->reviewer-sunets#error" data-target="edit-collection.reviewerSunetsField">
       <%= form.label :reviewer_sunets, 'Additional Reviewers', class: 'col-sm-2 col-form-label' %>
       <div class="col-sm-10 col-xl-8">
-        <%= form.text_field :reviewer_sunets, class: "form-control", required: false %> <!-- TODO: should prob have at least one required if enabled? -->
-        <div class="invalid-feedback">You must provide managers</div>
+        <%= form.text_field :reviewer_sunets, class: "form-control", required: false, data: { target: 'reviewer-sunets.container' } %> <!-- TODO: should prob have at least one required if enabled? -->
```
Comment (mjgiarlo) [inline app/components/collections/form_component.html.erb:104]: @peetucket here's the container you were asking about below, I believe

File: app/components/collections/form_component.rb [hunk 3]
```diff
@@ -1,12 +1,12 @@
-# typed: true
+# typed: false
```
Comment (mjgiarlo) [inline app/components/collections/form_component.rb:1]: :crying_cat_face:

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**